build.gradle

```diff
@@ -150,6 +150,16 @@ project(':iceberg-data') {
   }
 }
 
+project(':examples') {
```

Comment:
Maybe we could move the examples to the related modules, rather than gathering in a common module. I'm afraid if we're going to add Hive, Flink and other examples into this common module. This will really introduce the dependency problems.

Reply:
It is a very good point that dependency problems can become an issue if we collect all examples in one module. 
I've fixed all the other issues pointed out in the review and I can move tests around too if that's decided as the better approach?

Reply:
I think moving examples to dedicated modules would be better.